src/hotspot/share/code/SCCache.cpp

```diff
@@ -2289,6 +2481,11 @@ bool SCCache::store_exception_blob(CodeBuffer* buffer, int pc_offset) {
     buffer->decode();
   }
 #endif
+  // we need to take a lock to stop C1 and C2 compiler threads racing to
+  // write blobs in parallel with each other or with later nmethods
+  // TODO - maybe move this up to selected callers so we only lock
+  // when saving a c1 or opto blob
+  MutexLocker ml(Compile_lock);
```

Comment:
Only one compiler thread should generate compiler's runtime blobs.
See `AbstractCompiler::should_perform_init()`.

Reply:
Unless you changed when you call this generation.

Reply:
Done